docs/server/diagnostics/best-practices.md

```diff
@@ -0,0 +1,233 @@
+---
+title: "Best Practices"
+order: 4
+---
+
+# Monitoring Best Practices
+
+## Background
+
+When monitoring the health of a KurrentDB cluster, one should investigate and alert on multiple factors. Here we discuss them in detail
+
+## Metrics-based Monitoring
+
+The items in this section can be monitored using metrics from KurrentDB's Prometheus endpoint / Grafana dashboard, or via metrics from the operating system / machine / cloud provider
+
+### IOPS
+
+One should **monitor IOPS usage** to ensure it does not increase over 80% of allocation. This should take place at the operating system or machine level
+
+One should also **evaluate IOPS bursts** during extremely heavy periods, start of day / week, etc. to ensure they do not cause exhaustion. This should take place at the operating system or machine level
+
+Finally, **monitoring reader queue lengths** would help organizations understand if IOPS are exhausted as these queues will continue to grow in length, meaning the server never catches up with all read requests. This should take place using Kurrent's Grafana Dashboard
+
+![Reader Queue Lengths](./images/1-reader-queue-lengths.png)
+
+**At the first signs of IOPS exhaustion**, customers are advised to increase their IOPS limits
+
+### Memory Utilization
+
+As a database, we seek to use memory efficiently for improved processing. Organizations should perform a memory capacity confirmation test to establish baseline utilization, and monitoring should be performed to look for deviations from this baseline.
+
+Further, monitoring at the operating system level to ensure that **memory utilization does not exceed 85% of physical memory** helps mitigate allocation exceptions
+
+### Garbage Collection Pauses
+
+Garbage collection monitoring is largely concerned with gen2 memory, where longer-lived objects are allocated. The length of **application pauses for compacting garbage collection** of this generation should be monitored using the Kurrent Grafana Dashboard. Steadily increasing durations may eventually cause a leader election as the database will be unresponsive to heartbeats during compacting garbage collections. Monitor this metric to be below the configured Heartbeat Timeout value (default is 10 seconds, so for most customers, 8 seconds should be appropriate)
```

Comment:
I would suggest adding a link to this, since I don't believe the notion of "gen2 memory" is anything but a .NET implementation detail: https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/fundamentals.

Reply:
I might also add some commentary on leader election:  imagine that causes a short pause in the ability to perform steady writes?